File: ruby/grade-school/grade_school.rb
class School
	def initialize
		@roster = Hash.new
	end
	def students(grade)
		if @roster[grade] == nil
			return []
		end
		@roster[grade].sort
	end
	def add(name, grade)
		if @roster[grade] == nil
			@roster[grade] = [name]
		else
			@roster[grade].push(name)
		end
	end
	def students_by_grade
		all_students = []
		@roster.sort.each do |key, value|
			all_students.push({:grade=>key, :students=>value.sort})
		end
		all_students
	end
end
module BookKeeping
	VERSION = 3
end

File: javascript/grade-school/grade-school.js
var School = function(){
	this.school_roster = {};
	this.roster = function(){
		return this.school_roster;
	}
	this.add = function(name, grade){
		try{
			this.school_roster[grade].push(name);
		}
		catch(err){
			this.school_roster[grade] = [name];
		}
		this.school_roster[grade].sort();
	}
	this.grade = function(grade_number){
		if(this.school_roster[grade_number]){
			return this.school_roster[grade_number];
		}
		return [];
	}
};
module.exports = School;